.github/workflows/sync-ref.yml

```diff
@@ -0,0 +1,100 @@
+name: sync-ref
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: The ref to synchronize from git/git to gitgitgadget/git
+        type: string
+        default: refs/heads/master
+
+env:
+  SOURCE_REPOSITORY: git/git
+  TARGET_REPOSITORY: gitgitgadget/git
+  REF: ${{ inputs.ref || 'refs/heads/master' }}
+
+jobs:
+  sync-ref:
+    runs-on: ubuntu-latest
+    steps:
+      - name: check whether the ref is in sync
+        uses: actions/github-script@v6
+        id: check
+        with:
+          script: |
+            const getSHA = async (repository, ref) => {
+              if (ref.startsWith('refs/heads/') || ref.startsWith('refs/tags/')) ref = ref.substring(4)
+              else throw new Error(`Cannot handle ref '${ref}`)
+
+              try {
+                const { data: { object: { sha } } } = await github.rest.git.getRef(
+                  repository
+                    .split('/')
+                    .reduce((map, e) => {
+                      map[map.owner ? 'repo' : 'owner'] = e
+                      return map
+                  }, { ref })
+                )
+                return sha
+              } catch (e) {
+                if (e?.status == 404) return undefined
+                throw e
+              }
+            }
+
+            const sourceSHA = await getSHA(process.env.SOURCE_REPOSITORY, process.env.REF)
+            const targetSHA = await getSHA(process.env.TARGET_REPOSITORY, process.env.REF)
+            // skip sync if SHAs match, making extra certain that `master` is also synced to `main`
+            const skip = sourceSHA !== targetSHA
+              ? false
+              : (process.env.REF !== 'refs/heads/master' ||
+                 sourceSHA === await getSHA(process.env.TARGET_REPOSITORY, 'refs/heads/main'))
+            core.setOutput('skip', skip ? 'true' : 'false')
+            core.setOutput('source-sha', sourceSHA)
+            core.setOutput('target-sha', targetSHA)
```

Comment:
Should the second parm be `sha || ''` to ensure setOutput does not suddenly dislike undefined in the future?

Reply:
Excellent suggestion!